### ISSUE TITLE
gff_diff apollo flag

### ISSUE THREAD

**Issue by obsidian83** [sanger-pathogens/gff_diff#2]
when running gff_diff with Apollo flag it returns 

[src/bin/gff_diff.rs:59] apollo_diff(files[0].clone(), apollo_file) = Object(
    Object {
        store: [
            (
                "changes",
                Array(
                    [],
                ),
                0,
                0,
            ),
        ],
    },
)

without the flag it prints both Apollo and original GFF in JSON format

**Comment by magnusmanske** [comment]
Please pull the current master and try again, my internal PR was ignored by my colleagues so I merged it myself 😂

**Comment by obsidian83** [comment]
I've updated and tried again but the apply doesn't seem to be completely working, here are the outputs, only snoRNA line was augmented in the output

(base) Davids-MacBook-Pro:Tests davidstarns$ cat original.gff 
Tb927_09_v5.1	chado	gene	1918384	1918437	.	+	.	ID=Tb927_09_v4.snoRNA.0032
Tb927_09_v5.1	chado	snoRNA	1918384	1918437	.	+	.	ID=Tb927_09_v4.snoRNA.0032:snoRNA;Parent=Tb927_09_v4.snoRNA.0032;previous_systematic_id=Tb09_snoRNA_0092:snoRNA;product=term%3DC/D snoRNA%3Bdb_xref%3DPMID:15840815%3B;synonym=TB9Cs3C3.3
Tb927_09_v5.1	chado	CDS	1918384	1918437	.	+	0	ID=Tb927_09_v4.snoRNA.0032:exon:1;Parent=Tb927_09_v4.snoRNA.0032:snoRNA

(base) Davids-MacBook-Pro:Tests davidstarns$ cat apollo.gff 
Tb927_09_v5.1	.	gene	1918316	1918440	.	+	.	owner=D.E.Starns@liverpool.ac.uk;ID=198b60f3-0b74-40ab-adeb-0b5248d54f3f;date_last_modified=2019-08-21;Name=Tb927_09_v4.snoRNA.0032:snoRNA;date_creation=2019-08-21
Tb927_09_v5.1	.	snoRNA	1918316	1918440	.	+	.	owner=D.E.Starns@liverpool.ac.uk;Parent=198b60f3-0b74-40ab-adeb-0b5248d54f3f;ID=33d377ca-f11e-4433-a21e-bc186a8a4e61;orig_id=Tb927_09_v4.snoRNA.0032:snoRNA;date_last_modified=2019-08-21;Name=Tb927_09_v4.snoRNA.0032:snoRNA-00001;date_creation=2019-08-21
Tb927_09_v5.1	.	exon	1918316	1918440	.	+	.	Parent=33d377ca-f11e-4433-a21e-bc186a8a4e61;ID=0fbb113b-2147-4729-b57a-b21d62a969dd;Name=0fbb113b-2147-4729-b57a-b21d62a969dd

(base) Davids-MacBook-Pro:Tests davidstarns$ /Users/davidstarns/gff_diff/target/release/gff_diff --apollo --apply original.gff apollo.gff 
Tb927_09_v5.1	chado	CDS	1918384	1918437	.	+	0	ID=Tb927_09_v4.snoRNA.0032:exon:1;Parent=Tb927_09_v4.snoRNA.0032:snoRNA
Tb927_09_v5.1	chado	gene	1918384	1918437	.	+	.	ID=Tb927_09_v4.snoRNA.0032
Tb927_09_v5.1	chado	snoRNA	1918316	1918440	.	+	.	synonym=TB9Cs3C3.3;Parent=Tb927_09_v4.snoRNA.0032;ID=Tb927_09_v4.snoRNA.0032:snoRNA;previous_systematic_id=Tb09_snoRNA_0092:snoRNA;product=term%3DC/D snoRNA%3Bdb_xref%3DPMID:15840815%3B
(base) Davids-MacBook-Pro:Tests davidstarns$ /Users/davidstarns/gff_diff/target/release/gff_diff --apollo original.gff apollo.gff 
{
  "changes": [
    {
      "action": "update",
      "id": "Tb927_09_v4.snoRNA.0032:snoRNA",
      "key": "start",
      "value": "1918316",
      "what": "row"
    },
    {
      "action": "update",
      "id": "Tb927_09_v4.snoRNA.0032:snoRNA",
      "key": "end",
      "value": "1918440",
      "what": "row"
    }
  ]

**Comment by magnusmanske** [comment]
The snoRNA is the only entry in the Apollo GFF with an "orig_id", which means it's the only one I can _reliably_ trace back to the original GFF.

We have discussed adding a "guessing mode" based on chromosome/start/end/type, but it seemed not quite reliable enough.

**Comment by obsidian83** [comment]
Yeah I don't think guessing mode would be appropriate here, some snoRNAs were wildly miss annotated with some not having any overlap between previous and corrected. Hmm

**Comment by magnusmanske** [comment]
I added some inference (via parent/child and type) to find more original rows (now in master).

In your examples, it finds the gene and the snoRNA, but not the CDS since its type has changed to "exon".
